app/services/profiles/update.rb

```diff
@@ -1,52 +1,102 @@
 module Profiles
   class Update
-    def self.call(profile, updated_attributes = {})
-      new(profile, updated_attributes).call
+    include ImageUploads
+
+    def self.call(user, updated_attributes = {})
+      new(user, updated_attributes).call
     end
 
-    attr_reader
+    attr_reader :error_message
 
-    def initialize(profile, updated_attributes)
-      @profile = profile
-      @updated_attributes = updated_attributes
+    def initialize(user, updated_attributes)
+      @user = user
+      @profile = user.profile
+      @updated_profile_attributes = updated_attributes[:profile] || {}
+      @updated_user_attributes = updated_attributes[:user].to_h || {}
       @success = false
     end
 
     def call
+      unless verify_profile_image && update_profile && sync_to_user
+        Honeycomb.add_field("error", @error_message)
+        Honeycomb.add_field("errored", true)
+      end
+      self
+    end
+
+    def success?
+      @success
+    end
+
+    private
+
+    def verify_profile_image
+      image = @updated_user_attributes[:profile_image]
+      return true unless image
+      return true if valid_image_file?(image) && valid_filename?(image)
+
+      false
+    end
+
+    def valid_image_file?(image)
+      return true if file?(image)
+
+      @error_message = IS_NOT_FILE_MESSAGE
+      false
+    end
+
+    def valid_filename?(image)
+      return true unless long_filename?(image)
+
+      @error_message = FILENAME_TOO_LONG_MESSAGE
+      false
+    end
+
+    def update_profile
       # Ensure we have up to date attributes
       Profile.refresh_attributes!
 
       # Handle user specific custom profile fields
-      custom_attributes = @updated_attributes.extract!(*@profile.custom_profile_attributes)
-      @updated_attributes[:custom_attributes] = custom_attributes
+      if (custom_profile_attributes = @profile.custom_profile_attributes).any?
+        custom_attributes = @updated_profile_attributes.extract!(*custom_profile_attributes)
+        @updated_profile_attributes[:custom_attributes] = custom_attributes
+      end
 
       # We don't update `data` directly. This uses the defined store_attributes
       # so we can make use of their typecasting.
-      @profile.assign_attributes(@updated_attributes)
+      @profile.assign_attributes(@updated_profile_attributes)
 
       # Before saving, filter out obsolete profile fields
       @profile.data.slice!(*Profile.attributes)
 
-      return unless @profile.save
-
-      # Propagate changes back to the `users` table
-      sync_to_user
-      self
-    end
-
-    def success?
-      @success
+      @profile.save
     end
 
-    private
-
+    # Propagate changes back to the `users` table
```

Comment:
@benhalpern this will probably useful for you to see in terms of how we sync the profile data back to the user table. 

In addition, regarding backfilling, @citizen428 wrote a data_update script (which can be found in lib⁩/data_update_scripts /⁩20200819025131_migrate_profile_data.rb) that migrates the profile data for older users. You'll also notice a script BackfillProfileSkillsLanguages in this PR that does the same for a field that was missed.